src/ebonite/repository/metadata/local.py

```diff
@@ -463,6 +481,8 @@ def update_environment(self, environment: RuntimeEnvironment) -> RuntimeEnvironm
         return environment
 
     def delete_environment(self, environment: RuntimeEnvironment):
+        if self.data.get_instances_by_environment_id(environment.id):
+            raise EnvironmentWithInstancesError
```

Comment:
May be it's worth adding `environment` as parameter?

Reply:
Where exactly?

Reply:
`raise EnvironmentWithInstancesError(environment)`